pkg/agent/datapath/id_generate.go

```diff
@@ -0,0 +1,42 @@
+package datapath
+
+import (
+	"errors"
+	"sync"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/contiv/ofnet/ofctrl/cookie"
+)
+
+//nolint
+type IDGenerate struct {
+	lock     sync.RWMutex
+	idUint32 uint32
+	idUint64 uint64
+}
+
+//nolint
+func (i *IDGenerate) ascendUint32() {
+	i.lock.Lock()
+	defer i.lock.Unlock()
+
+	i.idUint32++
+}
+
+func (i *IDGenerate) ascendUint64() {
+	i.lock.Lock()
+	defer i.lock.Unlock()
+
+	i.idUint64++
+}
+
+var LearnCookieID = &IDGenerate{}
```

Comment:
It is ok to let `LearnCookieID` and `IDGenerate` be unexported?

Reply:
done